packages/api/src/utils/logParser.ts

```diff
@@ -247,17 +245,26 @@ export type VectorMetric = {
   v: number; // value
 };
 
-abstract class ParsingInterface<T> {
-  abstract _parse(
-    log: T,
-    ...args: any[]
-  ): Promise<LogStreamModel | MetricModel | RrwebEventModel>;
+const convertToStringMap = (blob: JSONBlob) => {
```

Comment:
Can we leave non-type changes to another PR ? What is this change about ? Any chance we can reuse `mapObjectToKeyValuePairs` ?

Reply:
No, this assignment needs a Record<string, string> and MOTKVP™ returns a bunch of arrays. I would end up doing a bunch of other rework that would be less clean, I think. Happy to take suggestions though.

This is essentially a bugfix revealed by the type changes - we were assigning an `any` (`metric.b`) to a `Record<string, string>` (`_string_attributes`) even though it could have any kind of value, so trying to type it left a type error here.

Reply:
Got it. although I'd prefer to get rid of the redundancy (maybe we can break down mapObjectToKeyValuePairs) but no big deal in the PR. Can we add a few unit tests for this method ?

Reply:
Yeah, you're right, the redundancy is no bueno. I've rewritten it to use MOTKVP and iterate over the properties to include bool/number values if they exist (though they should not, of course)